bin/middlewares/user_language.js

```diff
@@ -7,7 +7,14 @@ module.exports = function (languageConfig) {
   const supportedLanguages = languageConfig.supportedLanguages;
   return function (req, res, next) {
     const langHeaders = req.acceptsLanguages();
-    if (langHeaders) {
+
+    // If cookie "l" is set and corresponds to a supported language, use it
+    if (req.cookies && req.cookies.l) {
+      res.locals.language = supportedLanguages.find(lang => lang.code === req.cookies.l);
+    }
+
+    // Else, fallback to the first accepted language in the headers
+    else if (langHeaders) {
```

Comment:
Don't you think that `accept-language` header could still be used, if `l` is not a supported language ?

Reply:
yes, of course, and as far as I understand, `langHeaders` represents that header. That's why I use it in my "else", as a fallback.
sorry if I made a mistake here, I don't see the problem you describe :)

Reply:
@xem the problem is you will never pass in your `else if` condition if a `l` cookie was detected but the array `find` didn't match its value with a supported language.